ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/get_anta_results.py

```diff
@@ -0,0 +1,174 @@
+      save_catalog_name (bool): When set, the generated catalog is saved to a file using this name.
+      dry_run (boolean): if True, no test is actually run, useful in conjunction with save_catalog_name
+    """
+    if not HAS_ANTA:
+        raise AristaAvdError("AVD could not import the required 'anta' Python library")
+    if not HAS_YAML:
+        raise AristaAvdError("AVD could not import the required 'Pyaml' Python library")
+
+    # Setup ANTA logging
+    setup_logging(level=anta_logging)
+
+    if skipped_tests:
+        LOGGER.warning("The variable 'skipped_tests' has been set. Ansible tags are ignored for filtering tests.")
+    # Backward compatibility with eos_validate_state Ansible tags
+    elif ansible_tags:
+        # Retrieve the run_tags
+        run_tags = ansible_tags.get("ansible_run_tags", [])
+        # Retrieve the skip_tags
+        skip_tags = ansible_tags.get("ansible_skip_tags", [])
+        # Update the skipped_ variable according to legacy tags
+        skipped_tests = _get_skipped_tests_from_tags(run_tags, skip_tags)
+
+    # Create the ANTA catalog object with the appropriate skipped tests if any
+    catalog = Catalog(device_name, hostvars, skipped_tests=skipped_tests)
+
+    # NOTE: For now we only parse the default catalog
+    tests = parse_catalog(catalog.catalog)
+
+    if save_catalog_name is not None:
+        dumpable_ = {}
+        for test in tests:
+            test_class = test[0]
+            test_params = test[1]
+            dumpable_.setdefault(test_class.__module__, []).append({test_class.name: test_params})
+        with open(save_catalog_name, "w", encoding="UTF-8") as fd:
+            yaml.dump(dumpable_, fd, Dumper=NoAliasDumper)
+
+    # Create the ANTA ResultManager object
+    manager = ResultManager()
+
+    if dry_run:
+        LOGGER.info("DRY-RUN - generating empty results")
+        _create_dry_run_report(device_name, tests, manager)
+    else:
+        # Create the ANTA device object with the HttpApi connection session
+        device = AnsibleEOSDevice(name=device_name, connection=connection)
+        # Create the ANTA Inventory object and add the AnsibleEOSDevice device to it
+        inventory = AntaInventory()
+        inventory.add_device(device)
+
+        # Run
+        run(anta_runner(manager, inventory, tests))
+
+    # Save the results
+    results = loads(manager.get_results(output_format="json"))
+
+    # Format the data properly for the eos_validate_state report
+    for item in results:
+        item["categories"] = ", ".join(
+            [" ".join([word.upper() if word.lower() in ACRONYM_CATEGORIES else word.title() for word in category.split()]) for category in item["categories"]]
+        )
+
+        item["messages"] = "\n".join(item["messages"]).replace(",", "-")
+
+    return results
+
+
+def _create_dry_run_report(device_name: str, tests: list[tuple], manager: ResultManager) -> None:
+    """ """
```

Comment:
?

Reply:
what's the question? This is used to generate a report when using the dry_run options with all tests that would be run.
(check mode in ansible) - we have discussed this before.

Is it about the docstring?